src/main/java/reposense/git/GitRevList.java

```diff
@@ -57,15 +57,28 @@ public static String getCommitHashInRange(String root, String branchName, String
             return startHash;
         }
 
-        Path rootPath = Paths.get(root);
-        String revListCommand = "git rev-list " + startHash + ".." + endHash + " " + branchName
-                + REVISION_PATH_SEPARATOR;
-        String output = runCommand(rootPath, revListCommand);
+        String fromStartHash = getAllCommitHashSince(root, branchName, startHash);
```

Comment:
what happens if the start hash = end hash? might need to handle

Reply:
This is implicitly handled later on when computing the set difference, due to the additional step where we add back one of the hashes. If both sets of commits are the same, then the set difference will generate an empty list, then the original hash is added to the list again to output only the given hash.